### PR TITLE
Start testing against jruby

### PR DESCRIPTION
**Pull request by jyaworski**
Start testing against jruby per http://juanitofatas.com/2015/02/25/travis-ruby-template/

Also, use the latest ruby versions instead of pinning to a specific one.

@daenney @igalic @rnelson0

### CODE REVIEW

**Comment by rnelson0** [comment]
What benefit does jruby provide us, and does it outweigh the increased time for tests to complete with the additional test matrix entries? It's a 30% increase in tests and probably at least a 30% increase in time, unless jruby is faster.

**Comment by electrical** [comment]
There is no real advantage of testing against jruby. It need to start the JVM so by default its slower.

**Comment by jyaworski** [comment]
Puopetserver uses jruby. We should test against it to be complete. I myself have run into problems with jruby 1.9.3 vs ruby 1.9.3.

File: moduleroot/.travis.yml [hunk 1]
```diff
@@ -21,7 +21,10 @@ addons:
     <%- end -%>
   <%- end -%>
 <%- end -%>
-before_install: rm Gemfile.lock || true
+before_install:
+  - rm Gemfile.lock || true
+  - gem update --system || true
```
Comment (daenney) [inline moduleroot/.travis.yml]: Why?

File: moduleroot/.travis.yml [hunk 2]
```diff
@@ -21,7 +21,10 @@ addons:
...
+  - gem --version || true
```
Comment (daenney) [inline moduleroot/.travis.yml]: Why?

**Comment by rnelson0** [comment]
Why is 2.2 now tested against 4.0 but not 3.0, but 2.3 is tested against only 3.0? Perhaps you should re-sort the matrix by either puppet version, or ruby version, rather than the jumble it is now, to make it clearer what is and is not being tested.

File: moduleroot/.travis.yml (hunk 2 above)
Comment (jyaworski) [inline moduleroot/.travis.yml]: I could likely do without this; it was a best-practices thing I read.

File: moduleroot/.travis.yml (hunk 1 above)
Comment (jyaworski) [inline moduleroot/.travis.yml]: I could likely do without this; it was a best-practices thing I read.

**Comment by jyaworski** [comment]
@rnelson0 is there a canonical 'we should test against this' list?

**Comment by rnelson0** [comment]
Let's discuss the matrix questions in #113 and restrict comments here to the code review.

**Pull request merged**